File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        boolean answer = true;
        
        Stack<Character> st = new Stack<>();
        for (int i = 0; i < s.length(); i++) {
            char a = s.charAt(i);
            
            if (a == '(') {
                st.add(a);
            } else {
                if (st.empty() || st.peek() != '(') {
                    return false;
                }
                st.pop();
            }
        }

        return st.empty();
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.cpp
#include<string>
#include <iostream>
#include <stack>
using namespace std;

bool solution(string s){
    stack<char> st;
    for(int i=0;i<s.length();i++){
        if(s[i]==')'){
            if(st.empty()||st.top()!='('){
                return false;
            }
            st.pop();
            continue;
        }
        st.push(s[i]);
    }
    return st.empty();
}